### PR TITLE
Push down jsonb -> and ->> operators as ClickHouse dot notation

### PR DESCRIPTION
**Pull request by iskakaushik**
## Summary

- Rewrite jsonb `->` and `->>` operators to ClickHouse dot notation
  during deparse (e.g. `data ->> 'name'` becomes `data.name`), enabling
  ClickHouse to filter on JSON fields instead of shipping entire rows
  for local evaluation.
- Keys are emitted through `quote_identifier`, correctly quoting spaces,
  dots, mixed case, reserved words, and embedded double quotes.
- Follows up on bbdb1a6 which disabled subscript (`column['key']`)
  pushdown; this provides the working alternative via `->>`/`->`.

## Test plan

- [x] `EXPLAIN` verifies WHERE equality pushdown: `data.name = 'widget'`
- [x] `EXPLAIN` verifies WHERE LIKE pushdown: `data.name LIKE 'wid%'`
- [x] `EXPLAIN` verifies compound WHERE (AND / OR) pushdown
- [x] `EXPLAIN` verifies ORDER BY pushdown: `ORDER BY data.size`
- [x] `EXPLAIN` verifies `->` operator pushdown (returns jsonb, not text)
- [x] Edge cases: keys with spaces, mixed case, reserved words, embedded
  double quotes, backslashes, dots, apostrophes, special characters, and
  leading digits -- all produce correctly quoted remote SQL
- [x] Data fetches return correct values for all edge cases
- [x] All 4 expected output variants updated (PG 14-18, PG 13, CH 24.8-25.3, CH 22-24.3)
- [x] `make tempcheck` passes (json test green; pre-existing failures in
  functions, http, http_inserts, import_schema are unrelated)

### CODE REVIEW

File: src/custom_types.c [hunk 1]
```diff
@@ -380,6 +380,10 @@ classify_builtin_operator(const char *oprname)
 		return CF_REGEX_ICASE_MATCH;
 	if (strcmp(oprname, "!~*") == 0)
 		return CF_REGEX_ICASE_NO_MATCH;
+	if (strcmp(oprname, "->") == 0)
+		return CF_JSONB_FETCHVAL;
+	if (strcmp(oprname, "->>") == 0)
+		return CF_JSONB_FETCHVAL_TEXT;
```
Comment (theory) [inline src/custom_types.c]: Can we switch from these special constants to a table that maps operators? Something similar to `chfdw_check_for_custom_function`'s handling of functions.

File: src/deparse.c [hunk 2]
```diff
@@ -2745,6 +2745,35 @@ deparseOpExpr(OpExpr * node, deparse_expr_cxt * context)
 					goto cleanup;
 				}
 				break;
+			case CF_JSONB_FETCHVAL:
+			case CF_JSONB_FETCHVAL_TEXT:
```
Comment (theory) [inline src/deparse.c:2749]: These are different; I think `CF_JSONB_FETCHVAL` needs to convert results to JSON strings for compatibility.

File: test/expected/json.out [hunk 3]
```diff
@@ -136,6 +136,246 @@ SELECT data['size'], count(*) FROM json_http.things GROUP BY data['size'];
...
 
+-- The jsonb ->> operator should be pushed down to ClickHouse using dot
+-- notation (e.g. data ->> 'name' becomes data.name on the remote side).
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT data ->> 'name' FROM json_http.things;
+                   QUERY PLAN                    
+-------------------------------------------------
+ Foreign Scan on json_http.things
+   Output: (data ->> 'name'::text)
+   Remote SQL: SELECT data FROM json_test.things
```
Comment (theory) [inline test/expected/json.out:149]: Comment suggests this should be pushed down, but it's not (and won't be). Suggest removing it or commenting that this is expected.

File: test/expected/json.out [hunk 4]
```diff
@@ -136,6 +136,246 @@ SELECT data['size'], count(*) FROM json_http.things GROUP BY data['size'];
...
+(3 rows)
+
+-- The jsonb -> operator should also be pushed down using dot notation.
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT data -> 'name' FROM json_http.things;
+                   QUERY PLAN                    
+-------------------------------------------------
+ Foreign Scan on json_http.things
+   Output: (data -> 'name'::text)
+   Remote SQL: SELECT data FROM json_test.things
```
Comment (theory) [inline test/expected/json.out:227]: Same comment as above that this lack of pushdown is expected.

Comment (theory) [reply test/expected/json.out:227]: But also, there is no example here of `->` pushdown. If there were I suspect it would not work correctly, because `->` expects a JSON value, whereas the [dot notation](https://clickhouse.com/docs/sql-reference/data-types/newjson#reading-json-paths-as-sub-columns) seems to return strings. That means that something like this wouldn't find a record:

```sql
SELECT * FROM json_http.things WHERE data -> 'name' = '"widget"'::json;
```

File: test/expected/json.out [hunk 5]
```diff
@@ -136,6 +136,246 @@ SELECT data['size'], count(*) FROM json_http.things GROUP BY data['size'];
...
+(2 rows)
+
+-- Key with mixed case.
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT * FROM json_http.special_keys WHERE data ->> 'CamelCase' = 'world';
+                                           QUERY PLAN                                           
+------------------------------------------------------------------------------------------------
+ Foreign Scan on json_http.special_keys
+   Output: id, data
+   Remote SQL: SELECT id, data FROM json_test.special_keys WHERE ((data."CamelCase" = 'world'))
```
Comment (theory) [inline test/expected/json.out:325]: Nice.

**Review by theory — CHANGES_REQUESTED** [review]
This is great. `->`, however, does not work properly. We need to cast its output to JSON. Is that possible? IIRC ClickHouse only supports objects, but surely they can be nested.

File: test/expected/json.out [hunk 6]
```diff
@@ -136,6 +136,259 @@ SELECT data['size'], count(*) FROM json_http.things GROUP BY data['size'];
...
+-- WHERE clause with -> wraps the dot notation in toJSONString() so that the
+-- result is a proper JSON value (-> returns jsonb, not text like ->>).
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT * FROM json_http.things WHERE data -> 'name' = '"widget"'::jsonb;
+                                             QUERY PLAN                                             
+----------------------------------------------------------------------------------------------------
+ Foreign Scan on json_http.things
+   Output: id, data
+   Remote SQL: SELECT id, data FROM json_test.things WHERE ((toJSONString(data.name) = '"widget"'))
+(3 rows)
```
Comment (theory) [inline test/expected/json.out:282]: Would like to see the actual output from this query, which is not currently run on ClickHouse. Would be useful to have it return a few types — boolean, number, string, object — just to ensure that we get the expected results.

**Review by theory — APPROVED** [review]
Looks good, just would like a few more `->` tests to ensure `toJSONString()` does what we think it does.

After that, please rebase and create a single commit that describes the changes. Thank you!

File: test/expected/json.out (hunk 4 above)
Comment (theory) [reply test/expected/json.out:227]: Added in 0b4c03e.

File: test/expected/json.out (hunk 6 above)
Comment (theory) [reply test/expected/json.out:282]: Added in 0b4c03e.
